src/protean/impl/repository/dict_repo.py

```diff
@@ -49,28 +50,72 @@ def _create_object(self, model_obj):
 
         return model_obj
 
-    def _filter_objects(self, page: int = 1, per_page: int = 10,
-                        order_by: list = (), _excludes=None, **filters):
-        """ Read the repository and return results as per the filer"""
-
-        # Filter the dictionary objects based on the filters
-        items = []
-        excludes = _excludes if _excludes else {}
-        for item in self.conn['data'][self.model_name].values():
+    def _extract_lookup(self, key):
```

Comment:
This function can be moved to the `BaseAdapter` class and thus allowing to be overridden if needed

Reply:
Good idea. Moved.